command/certificate/sign.go

```diff
@@ -435,6 +448,19 @@ func createTemplateData(cr *x509.CertificateRequest, maxPathLen int) x509util.Te
 		sans = append(sans, v.String())
 	}
 
+	if !omitCNSAN && len(sans) == 0 {
+		cnInSANs := false
+		for _, v := range sans {
+			if v == cr.Subject.CommonName {
+				cnInSANs = true
+				break
+			}
+		}
+		if !cnInSANs {
+			sans = append(sans, cr.Subject.CommonName)
+		}
```

Comment:
len(sans) is 0, so we don't need to do this. We can just do the following:

```suggestion
		sans = append(sans, cr.Subject.CommonName)
```

We might want to see what happens if the common name is empty and probably add a check for that.